### PR TITLE
Setup CD for agent

### PR DESCRIPTION
**Pull request by gouthamve**
This deploys agent to dev on a merge to main.

Depends on https://github.com/grafana/deployment_tools/pull/19936

Currently this automation only opens a PR, but I want to double check things and then I'll deploy directly to dev without a PR.

Signed-off-by: Goutham Veeramachaneni <gouthamve@gmail.com>

### CODE REVIEW

File: .drone/drone.yml
Comment (rfratto) [inline .drone/drone.yml:144]: Haha, this makes me wish we were using sonnet to generate this.

**Review by rfratto — COMMENTED** [review]
LGTM, though you'll need to run `make drone` to sign it. If you don't have DRONE_TOKEN set up in your environment, you'll need to have that defined in advance.

File: .drone/drone.yml
Comment (rfratto) [inline .drone/drone.yml:143]: For the curious, these are [converted](https://docs.drone.io/plugins/overview/#plugin-inputs) into `PLUGIN_<settings key>` environment variables for the container to use.

File: .drone/drone.yml
Comment (jvrplmlmn) [inline .drone/drone.yml:144]: You can do this as a multi-line config, something like:

```
  settings:
    config_json: |-
      {
        "key1": "value1",
        "key2": "value2"
      }
```

**Comment by gouthamve** [comment]
@rfratto and @jvrplmlmn Thanks for the reviews, I've updated the PR! I think this is good to go, but I'll only know if the tags and images are sane once this is merged. Once I confirm that I'll remove the PR requirement and directly commit to `master`.

File: .drone/drone.yml
Comment (jvrplmlmn) [inline .drone/drone.yml:161]: If I understand https://docs.drone.io/pipeline/environment/reference/drone-tag/ correctly ,`DRONE_TAG` will be empty for this pipeline for some of the events, see the trigger section:

```
trigger:
  ref:
    - refs/heads/main
```

You are not filtering on a specific event (such as: `push`, `pull_request` or `tag`) so it would apply to all fo them.

File: .drone/drone.yml
Comment (jvrplmlmn) [inline .drone/drone.yml:169]: Looking at the whole `drone.yml` plus the Makefile, I think that this won't work and you should depend instead on `Release`. If I understand correctly, the `Containerize` builds but does not push the Docker image to the registry, that is done in `Release` via `make ... publish`.

Comment (rfratto) [reply .drone/drone.yml:169]: Containerize does push the images through dockerx, release is for non-container release assets :)

Comment (jvrplmlmn) [reply .drone/drone.yml:169]: Then you are golden 👍

Comment (gouthamve) [reply .drone/drone.yml:169]: Yep, I was looking at the logs and noticed that `Containerize` does push things.

File: .drone/drone.yml
Comment (gouthamve) [reply .drone/drone.yml:161]: Hrm, this is true. I need to somehow get `IMAGE_TAG ?= $(shell ./tools/image-tag)` in there. Not sure how. Looking at how other repos are doing it.

Comment (rfratto) [reply .drone/drone.yml:161]: I feel responsible for this being a pain so I opened grafana/deployment_tools#20048 to add `jsonnet_value_file` support. 

Once merged and the updated image gets deployed, we can use an initial stage to write the output of `$(shell ./tools/image-tag)` to a file and use the file name in the config_json instead.

Comment (rfratto) [reply .drone/drone.yml:161]: That PR has been merged and the docker image is updated to include the new feature.

**Comment by gouthamve** [comment]
So this works! And it directly commits to master and doesn't open a PR. Which is what we want for the dev rollouts.

File: .drone/drone.yml
Comment (rfratto) [inline .drone/drone.yml:159]: We can probably take this part out, but I think it's useful to have it in for when we want to copy this for non-dev flows

**Review by rfratto — APPROVED** [review]
Love it!

**Review by mattdurham — APPROVED** [review]
Looks reasonable

**Pull request merged**
